liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/AbstractLibertySupport.java

```diff
@@ -251,16 +257,18 @@ protected ArtifactItem createArtifactItem(String groupId, String artifactId, Str
      * @param artifactId String specifying the artifactId of the Maven artifact to copy.
      * @param version String specifying the version of the Maven artifact to copy.
      * @param type String specifying the type of the Maven artifact to copy.
+     * @param classifier String specifying the classifier of the Maven artifact to copy.
      *
      * @return Set<Artifact> A collection of Artifact objects for the resolved dependencies and transitive dependencies
      * @throws MojoExecutionException
      */
-    protected Set<Artifact> getResolvedDependencyWithTransitiveDependencies(String groupId, String artifactId, String version, String type) throws MojoExecutionException {
+    protected Set<Artifact> getResolvedDependencyWithTransitiveDependencies( String groupId, String artifactId, String version, String type,
+                                                                             String classifier ) throws MojoExecutionException {
         Set<Artifact> resolvedDependencies = new HashSet<Artifact> ();
 
         if (version != null) {
             // if version is set, it will always override the one in project dependency
-            Artifact artifact = getArtifact(groupId, artifactId, type, version);
+            Artifact artifact = getArtifact(groupId, artifactId, type, version, classifier);
```

Comment:
So is the `classifier` only going to be supported when the `version` is specified? And no wildcards for groupId and artifactId?

Reply:
Also, need to include the classifier in the log warning at line 276.

Reply:
At line 321 do we need to include the classifier?

Reply:
Added to the logging message.
About your question: 

> So is the classifier only going to be supported when the version is specified? 

Good. If no version is specified, it will try to use the dependencies from the main project, which may ambiguous (depending on how you want to treat classifiers)9. No one will prevent you from  adding the same dependency with different versions and different classifiers to a maven project. Each `g:a:v:c` can be unique. If you only have `g:a`, what will you choose? The only option can only be the dependency where `c == null` (i.e. not set).

If you only have `g:a::c` (no version), the only correct way should be to search for a coordinate with the same `g:a::c` in your dependencies. Will add that in a later commit.

It needs to be included inside THIS monster of a diff (which I surely will refactor out into a method and write a unit test for…):

```java
                if (artifact.getGroupId().equals(groupId) && 
                    ((compareArtifactId == null) ||
                     (isWildcard && artifact.getArtifactId().startsWith(compareArtifactId)) ||
                     (artifact.getArtifactId().equals(compareArtifactId)))) {
```

… although I see no unit tests. Is this intentional? 🤔 

Line 321: Yes, if I get this correctly, this is the line where no version was specified?

Reply:
Yes, no unit tests are intentional. We run integration tests for every build and test functionality the way the user would use it.

If we only want to honor the classifier when the version is specified, I think a check needs to be added somewhere. I'm not sure after your examples where you landed on that point. 

btw..we do have some existing ITs for copyDependencies...in `appsdirectory-notset-configured-it/pom.xml` and `appsdirectory-notset-notconfigured-it/pom.xml`.